File: framework/src/gmsec5/field/I64Field.cpp
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


#include <gmsec5/field/I64Field.h>
#include <gmsec5/internal/field/InternalI64Field.h>


using namespace gmsec::api5;
using namespace gmsec::api5::internal;


I64Field::I64Field(const char* name, GMSEC_I64 value, bool isHeader)
	: m_internal(new InternalI64Field(name, value, isHeader))
{
	registerChild(m_internal);
}


I64Field::I64Field(const I64Field& other)
	: m_internal(new InternalI64Field(other.getName(), other.getValue(), other.isHeader()))
{
	registerChild(m_internal);
}


I64Field::~I64Field()
{
	delete m_internal;
}


GMSEC_I64 I64Field::getValue() const
{
	return m_internal->getValue();
}


const char* I64Field::toXML() const
{
	return m_internal->toXML();
}


const char* I64Field::toJSON() const
{
	return m_internal->toJSON();
}

File: nodejs/gmsec5/dox/I64Field.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file I64Field.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain a signed 64-bit value.
 */
public class I64Field extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data A signed 64-bit value.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public I64Field(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      A signed 64-bit value.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public I64Field(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other I64Field object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given I64Field is null.
	 */
	public I64Field(other);


	/**
	 * Returns the data stored within the I64Field.
	 *
	 * @return A signed 64-bit value.
	 */
	public getValue();
}
